x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -234,6 +235,34 @@ static DeprecationIssue checkTemplatesWithMultipleTypes(ClusterState state) {
         );
     }
 
+    static DeprecationIssue checkTemplatesWithCustomTypes(ClusterState state) {
+        Set<String> templatesWithCustomTypes = new TreeSet<>();
+        state.getMetadata().getTemplates().forEach((templateCursor) -> {
+            String templateName = templateCursor.key;
+            ImmutableOpenMap<String, CompressedXContent> mappings = templateCursor.value.mappings();
+            if (mappings != null && mappings.size() == 1) {
```

Comment:
I think we should check the type name even if there is more than one mapping type. For example, a user could have both a `_default_` mapping and a custom one like `some_type`.

This makes me think -- maybe this check could be combined with the one above `checkTemplatesWithMultipleTypes`? That way we could report all errors at once (both that they are using a `_default_` mapping, and that they have an custom type name). The error message about "multiple types" could be improved to talk about a `_default_` mapping to be more specific.

Reply:
I was only checking for single mapping types with the thought that checkTemplatesWithMultipleTypes would catch the others. I'll see if I can find a clean way to combine them.

Reply:
My earlier comment was based on a misunderstanding -- I thought you couldn't have templates with multiple custom types in 7.x, but you actually can if they are carried over from 6.x! See https://github.com/elastic/elasticsearch/pull/72540 for details. I thought this was only possible when using the `_default_` mapping, but that is not true. So no need to add special handling for `_default_`.